frontend/catalyst/pennylane_extensions.py

```diff
@@ -126,11 +127,22 @@ def __call__(self, *args, **kwargs):
                 name = self.device.short_name
             else:
                 name = self.device.name
-            if name not in QFunc.RUNTIME_DEVICES:
+
+            is_known_device = name in QFunc.RUNTIME_DEVICES
+            implements_c_interface = hasattr(self.device, "get_c_interface")
```

Comment:
Should we subclass QJitDevice here instead of making if-branches?

Reply:
I don't think we want to subclass `QJITDevice`. It is my understanding we are trying to remove `QJITDevice` and keep the device as defined by the user (as long as they follow the new device specific and the ADR for how to integrate with devices).

But you have a good point that instead of rewriting the name, I can just add an optional field to QJITDevice (or similar).